lib/migrations/state.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * migrations/state.js - migration state for hsd.
+ * Copyright (c) 2021, Nodari Chkuaselidze (MIT License)
+ */
+'use strict';
+
+const assert = require('bsert');
+const bio = require('bufio');
+const {encoding} = bio;
+
+/**
+ * State of database migrations.
+ * because migration IDs are only increasing, we only need
+ * to store the last one.
+ * @alias module.migrations.MigrationState
+ * @property {Boolean} inProgress - is/was migration in progress
+ * NOTE: If inProgress is true, we know lastMigration + 1 was the one in
+ *       the progress.
+ * @property {Number} lastMigration - last migration
+ * NOTE: Migration numbers start from 1.
+ *       0 - means there is no migration.
+ */
+
+class MigrationState extends bio.Struct {
+  /**
+   * Create MigrationState
+   * @constructor
+   */
+
+  constructor() {
+    super();
+
+    this.inProgress = false;
+    this.lastMigration = 0;
+    this.skipped = [];
+  }
+
+  /**
+   * Inject properties from another state.
+   * @param {MigrationState} obj
+   * @returns MigrationState;
+   */
+
+  inject(obj) {
+    assert(obj instanceof MigrationState);
+    this.inProgress = obj.inProgress;
+    this.lastMigration = obj.lastMigration;
+    this.skipped = obj.skipped.slice();
+    return this;
+  }
+
+  /**
+   * Get size of the encoded migration state object.
+   * @returns {Number}
+   */
+
+  getSize() {
+    // flags + last migration number
+    let size = 4; // flags
+    size += encoding.sizeVarint(this.lastMigration); // last migration ID
```

Comment:
You don't wanna have a little confidence that we'll need less than 4,294,967,295 migrations and lock this in to a slim U32? Same with `skipped`...?

Reply:
Actually it's the reverse, I don't think it will get bigger than 253 (which will take 1 byte), even if it does it's highly unlikely to get bigger than 65k - We definitely have different problems at that point.

So up to 253 it will just take a byte instead of 4 and is still flexible for bigger numbers.